src/Microsoft.Azure.WebJobs.Host/Common/ITriggerBindingStrategy.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// Strategy pattern to describe how to bind a core trigger type to various parameter.
+    /// Suports:
+    /// - both Single-item vs. Batch dispatch.  
+    /// - core binding, string, poco w/ Binding Contracts
+    /// 
+    /// For example, a single EventHubTriggerInput -->  can bind to 
+    ///  EventData, EventData[], string, string[], Poco, Poco[]    
```

Comment:
This list isn't exhaustive, right? Can't it also bind to byte[] and other things? What defines the full list for EventHubTrigger?